src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs

```diff
@@ -19412,6 +19412,142 @@ public void AbstractInSubmission()
                 // internal abstract event System.EventHandler E;
                 Diagnostic(ErrorCode.ERR_AbstractInConcreteClass, "E").WithArguments("E", "Script").WithLocation(3, 45));
         }
+
+        [Fact, WorkItem(16484, "https://github.com/dotnet/roslyn/issues/16484")]
+        public void MultipleForwardsToTheSameAssemblyShouldIgnoreDuplicate()
+        {
+            var forwardedToCode = @"
+namespace Destination
+{
+    public class TestClass
+    {
+        public const string Value = ""TEST VALUE"";
+    }
+}";
+            var forwardedToReference = CreateCompilationWithMscorlib(forwardedToCode, assemblyName: "Destination").EmitToImageReference();
+
+            var forwardingCode = @"
+namespace ForwardingNamespace
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            System.Console.WriteLine(Destination.TestClass.Value);
+        }
+    }
+}";
+            var forwardingIL = @"
+.assembly extern Destination { }
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination
+}
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination
+}";
+            var forwardingCompilation = CreateCompilationWithCustomILSource(
+                source: forwardingCode,
+                ilSource: forwardingIL,
+                references: new MetadataReference[] { forwardedToReference },
+                options: new CSharpCompilationOptions(OutputKind.ConsoleApplication));
+
+            CompileAndVerify(forwardingCompilation, expectedOutput: "TEST VALUE");
+        }
+
+        [Fact, WorkItem(16484, "https://github.com/dotnet/roslyn/issues/16484")]
+        public void MultipleForwardsToDifferentAssembliesWithoutReferencingThemShouldIgnoreIt()
+        {
+            var forwardingIL = @"
+.assembly extern Destination1 { }
+.assembly extern Destination2 { }
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination1
+}
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination2
+}";
+            var forwardingCompilation = CreateCompilationWithCustomILSource(string.Empty, forwardingIL);
+
+            forwardingCompilation.VerifyDiagnostics();
```

Comment:
At the very least we should emit and verify the produced assembly. It would be good to have it print something so that we could also verify that we can run the code.